libsolidity/formal/CHC.cpp

```diff
@@ -112,72 +112,182 @@ vector<string> CHC::unhandledQueries() const
 {
-	if (!_function.isImplemented())
+	//THIS THING HERE NEEDS TO BE A NONLINEAR CLAUSE
+	//constructor(...) THAT INITIALIZES ITS STATE VARIABLES
+	//AND RUNS THE CONSTRUCTOR CODE, IF AVAILABLE
+	auto const& implicitConstructorPredicate = *createSymbolicBlock(
+		constructorSort(_contract, state()),
+		"contract_initializer_entry_" + contractSuffix(_contract) + "_" + uniquePrefix(),
+		PredicateType::ConstructorSummary,
+		&_contract
+	);
+
+	smtutil::Expression preConstraints = state().state() == state().state(0);
+	for (auto var: stateVariablesIncludingInheritedAndPrivate(_contract))
+		preConstraints = preConstraints && m_context.variable(*var)->valueAtIndex(0) == currentValue(*var);
+	if (auto constructor = _contract.constructor())
+		for (auto var: constructor->parameters())
+			preConstraints = preConstraints && m_context.variable(*var)->valueAtIndex(0) == currentValue(*var);
+	preConstraints = preConstraints && errorFlag().currentValue() == 0;
+	m_context.addAssertion(preConstraints);
+
+	auto implicitFact = smt::constructor(implicitConstructorPredicate, m_context);
+	//connectBlocks(m_currentBlock, implicitFact);
+	addRule(smtutil::Expression::implies(preConstraints, implicitFact), implicitFact.name);
+	setCurrentBlock(implicitConstructorPredicate);
+
+	for (auto var: _contract.stateVariables())
+		m_context.variable(*var)->increaseIndex();
+	initializeStateVariables(_contract);
+	auto const& afterInit = *createSymbolicBlock(
+		constructorSort(_contract, state()),
+		"contract_initializer_after_init_" + contractSuffix(_contract) + "_" + uniquePrefix(),
+		PredicateType::ConstructorSummary,
+		&_contract
+	);
+	connectBlocks(m_currentBlock, predicate(afterInit));
+	setCurrentBlock(afterInit);
+
+	if (auto constructor = _contract.constructor())
 	{
-		addRule(summary(_function), "summary_function_" + to_string(_function.id()));
-		return false;
+		errorFlag().increaseIndex();
+		m_context.addAssertion(smt::functionCall(*m_summaries.at(&_contract).at(constructor), &_contract, m_context));
+		connectBlocks(m_currentBlock, initializer(_contract), errorFlag().currentValue() > 0);
+		m_context.addAssertion(errorFlag().currentValue() == 0);
 	}
 
-	// This is the case for base constructor inlining.
-	if (m_currentFunction)
+	connectBlocks(m_currentBlock, initializer(_contract));
+}
+
+map<ContractDefinition const*, vector<ASTPointer<frontend::Expression>>> CHC::baseArguments(ContractDefinition const& _contract)
+{
+	map<ContractDefinition const*, vector<ASTPointer<Expression>>> baseArgs;
+
+	for (auto contract: _contract.annotation().linearizedBaseContracts)
 	{
-		solAssert(m_currentFunction->isConstructor(), "");
-		solAssert(_function.isConstructor(), "");
-		solAssert(_function.scope() != m_currentContract, "");
-		SMTEncoder::visit(_function);
+		/// Collect base contracts and potential constructor arguments.
+		for (auto inh: contract->baseContracts())
+		{
+			solAssert(inh, "");
```

Comment:
Can this happen at all?

Reply:
I am not suggesting to remove assert, it should be there, just thinking when and how `null` would be added to the list :)

Reply:
Yea I also don't know when it happens, but the function returns a pointer, so better test it before using it.